vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java

```diff
@@ -386,6 +386,9 @@ else if (OWNING_BOARD.equals(key)) {
           owningBoardName = null;
         }
       }
+      else if (value.equals("")) {
```

Comment:
You should  check `"".equals(value)` instead. This way, you're still safe if `value` is null.

Reply:
Fixed